@xen-orchestra/backups/_getOldEntries.test.mjs

```diff
@@ -0,0 +1,164 @@
+          { timestamp: +new Date('2024-07-04 00:09:00'), id: 3 }, // too old
+          { timestamp: +new Date('2024-08-12 00:09:00'), id: 4 }, // too old
+          { timestamp: +new Date('2024-09-05 00:09:00'), id: 5 }, // too old
+          { timestamp: +new Date('2024-10-02 00:09:00'), id: 6 }, // new month,
+          { timestamp: +new Date('2024-11-01 00:09:00'), id: 7 }, // new month , week reached retention
+          { timestamp: +new Date('2024-12-17 00:09:00'), id: 8 }, // new week
+          { timestamp: +new Date('2024-12-24 00:09:00'), id: 9 }, // new week/month / year  daily reach retention
+          { timestamp: +new Date('2025-01-01 00:09:00'), id: 10 }, // same day/week/month/year
+          { timestamp: +new Date('2025-01-01 00:10:00'), id: 11 }, //  new  day / week / month
+          { timestamp: +new Date('2025-12-31 00:09:00'), id: 12 }, // same day/week/month/year
+          { timestamp: +new Date('2025-12-31 00:09:00'), id: 13 }, // new month /year
+        ],
+        {
+          longTermRetention: {
+            daily: { retention: 2 },
+            weekly: { retention: 4 },
+            monthly: { retention: 5 },
+            yearly: { retention: 2 },
+          },
+          timezone: 'Europe/Paris', // use a time zone here because week definition is timezone bound
+        },
+      ],
+      expectedIds: [1, 2, 3, 4, 5, 10, 12],
+      testLabel: 'complete test  ',
+    },
+    {
+      args: [
+        0,
+        [
+          { timestamp: +new Date('2024-09-01 00:01:00'), id: 1 }, // thrid day too old
+          { timestamp: +new Date('2024-09-01 00:10:00'), id: 2 }, // thrid day
+          { timestamp: +new Date('2024-09-01 00:20:00'), id: 3 }, // second day, too old
+          { timestamp: +new Date('2024-09-02 00:22:00'), id: 4 }, // second day too old
+          { timestamp: +new Date('2024-09-03 00:20:00'), id: 5 }, // second day in NZ
+          { timestamp: +new Date('2024-09-04 00:09:00'), id: 6 }, // same day in NZ
+          { timestamp: +new Date('2024-09-04 00:10:00'), id: 7 }, // most recent
+        ],
+        {
+          longTermRetention: {
+            daily: { retention: 3 },
+          },
+          timezone: 'Pacific/Auckland', // GMT +6
+        },
+      ],
+      expectedIds: [1, 2, 3, 6],
+      testLabel: 'should  handle timezone ',
+    },
+  ]
+
+  for (const { args, expectedIds, testLabel } of tests) {
+    it(testLabel, () => {
+      const oldEntries = getOldEntries.apply(null, args)
+      assert.strictEqual(
+        oldEntries.length,
+        expectedIds.length,
+        `different length , ${JSON.stringify({ oldEntries, expectedIds })}`
+      )
+      for (let i = 0; i < expectedIds.length; i++) {
+        assert.strictEqual(
+          oldEntries[i].id,
+          expectedIds[i],
+          `different id , ${JSON.stringify({ i, expectedIds, oldEntries })}`
+        )
+      }
+    })
+  }
+})
+
+describe('_getOldEntries() should fail when called incorrectly', () => {})
```

Comment:
Why doesn't it do anything?

Reply:
The describe is still empty.

Reply:
I should missing something,  the 2 describes seems to have content from my point of view

Reply:
Yes I didn't see that it was outdated at the time of the comment.
Then I approved the PR but forgot to close the comment, I'll do that right away.